api/src/main/java/com/netflix/iceberg/Transaction.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.iceberg;
+
+import com.netflix.iceberg.exceptions.CommitFailedException;
+import com.netflix.iceberg.exceptions.ValidationException;
+
+/**
+ * A transaction for performing multiple updates to a table.
+ */
+public interface Transaction {
+  /**
+   * Return the {@link Table} that this transaction will update.
+   *
+   * @return this transaction's table
+   */
+  Table table();
```

Comment:
I might be getting ahead of myself but don't we want the transaction to eventually span across tables? in which case shouldn't we change this API so it can operate over multiple tables?

Reply:
I think we'd probably introduce a cross-table transaction API for that, and it isn't something we will do soon.

For now, we need to be able to get a table that creates updates that are part of the transaction. That way, callers can write code that appends to a table and that code doesn't need to know that the operation is part of a larger transaction.